GuideFinder-Mac/GuideFinder_Mac_2020_SMM_PAMless.Rmd

```diff
@@ -0,0 +1,1365 @@
+```{r}
+
+PromoterRegion<-0
+
+SubtractProm<-function(x) {
+  x - PromoterRegion
+}
+
+AddProm<-function(x) { 
+  x + PromoterRegion 
+}
+
+#subset the promoter sequence 
+SubsetPromoter<-function (x) {
+  substr(x, 0, PromoterRegion)
+}
+
+```
+
+
+##5.Complete vs Draft Genome
+Set option to run program on a draft genome (see Guide Finder Protocol document for detailed input instructions) or complete genome (using GenBank Accession #).     
+__For running program on a complete genome using GenBank Accession #, set CompleteGenone to TRUE; if using a draft genome set CompleteGenome to FALSE__
+```{r,tidy=TRUE}
+CompleteGenome<- TRUE
+```
+
+
+##5A.FOR COMPLETE GENOMES ONLY:
+Input GenBank Accession Number. Will retrieve the annotated genome from GenBank (called by accession #, e.g.AE015929.1 for *Staphylococcus epidermidis*).   
+From the annotated file, will retrieve the coding sequence information.  
+*If using a draft genome, do not run this chunk of code.*  
+__Example__: Substitute the bolded information for the GenBank Accession # for your genome of interest.   
+GenBankInfo<- readGenBank(GBAccession( __"AE014133.2"__ ))   
+```{r,tidy=TRUE}
+#retrieves annotation from GenBank, called by accession
+GenBankInfo<- readGenBank(GBAccession("CP017100.1")) 
+#from annotated file, retrieves coding sequence information
+CDSInfoDF<-data.frame(attr(GenBankInfo, "cds")) 
+```
+
+##5B.FOR DRAFT GENOMES ONLY: 
+Input pre-processed input file (containing product/gene name, strand type, and start/end coordinates)
+This loop takes the gene coordinates and modifies them to include the putative promoter region 
+__Example__: Substitute the bolded information for the name of your file which contains the gene coordinates (output from custom or supplied pre-processing scripts). File should be kept in the working directory.  
+NewCoordinatesDF<-fread( __"DrafGenome_NewCoordinates.csv"__ )  
+*if using a complete genome/GenBank Accession #, do not run this chunk of code*  
+```{r,tidy=TRUE}
+
+if (CompleteGenome == FALSE) {
+CoordinatesDF<-fread("DraftGenome_NewCoordinates.csv")
+NewCoordinatesDF<-NULL
+for (i in 1:nrow(CoordinatesDF)) { 
+  WorkingRow<-CoordinatesDF[i,]
+  Temp<-if (WorkingRow$strand == "-") {replace(WorkingRow,4,AddProm(WorkingRow$end))} #for genes coded on the minus strand, add bp to the end coordinate; capture putative promoter sequence 
+  else {replace(WorkingRow,3, SubtractProm(WorkingRow$start))} #for genes coded on the plus strand, subtract bp from the start coordinate; capture putative promoter 
+  NewCoordinatesDF<-rbind(NewCoordinatesDF,Temp)
+} }
+names(NewCoordinatesDF)<-c("geneproduct","strand", "start", "end")  #change column names to those that are used by the rest of the Guide Finder script
+
+```
+
+
+##6.Set Parameters for Guide Design
```

Comment:
Or should the promoter region length also be included in the defs of parameters here?